apps/web/src/pages/auth/InvitationPage.tsx

```diff
@@ -112,14 +110,23 @@ export default function InvitationPage() {
             ) : undefined
           }
         >
-          <LoadingOverlay
-            visible={isInitialLoading}
-            overlayColor={colors.B30}
-            loaderProps={{
-              color: colors.error,
-            }}
-          />
-          {!isInitialLoading && <SignUpForm email={data?.email} token={tokenParam} />}
+          {isInitialLoading ? (
+            <LoadingOverlay
+              visible
+              overlayColor={colors.B30}
+              loaderProps={{
+                color: colors.error,
+              }}
+            />
+          ) : (
+            <>
+              {existingUser ? (
+                <LoginForm email={data?.email} invitationToken={invitationToken} />
+              ) : (
+                <SignUpForm email={data?.email} invitationToken={invitationToken} />
```

Comment:
We can do something like:
```
const ClassForm = existingUser ? LoginForm : SignUpForm;

return (<ClassForm email={data?.email} invitationToken={invitationToken} />);
```

Reply:
sure, at the beginning I had different args there, but now it's the same, so can be changed I guess.